### ISSUE TITLE
Hardcoded ZMQ buffer sizes are brittle

### ISSUE THREAD

**Issue by thomasjm** [joeltg/mit-scheme-kernel#2]
This kernel has code like the following to decide on how much buffer to allocate for receiving ZMQ messages:

```
(define hb-length 4)
(define uuid-length 33)
(define delimiter-length (string-length delimiter))
(define signature-length 64)
(define header-length 256)
(define parent-length 256)
(define metadata-length 2)
(define content-length 4096)
```

Hardcoding these numbers is super brittle, because they are in no way guaranteed to be an upper bound on the actual sizes (see the spec here: http://jupyter-client.readthedocs.io/en/latest/messaging.html#the-wire-protocol). This caused weird bugs for a custom Jupyter client I'm writing because my UUIDs were slightly over 33 bytes. I'm guessing these numbers came from reverse-engineering the normal Python Jupyter messages?

I think the right thing to do is use `zmq_recvmsg` instead of `zmq_recv`, because the former looks like it takes care of allocating a suitably sized buffer...

**Comment by joeltg** [comment]
You're absolutely right - `zmq_recv` is the correct way to do this and hardcoding the values is terribly brittle. If I remember right, I wasn't able to get `zmq_recv` to work with MIT Scheme's foreign function interface - I'll take a look again (and post the errors if anyone has any clues).

**Comment by joeltg** [comment]
Fixed in 5d5d3e34ec6e69a260ce7528f57bba8a5f697814!
